Storage clear documentation (#2796)
Codecademy/docs: content/javascript/concepts/storage/terms/clear.md

---
Title: '.clear()'

> Description: '.clear() removes all keys in a Storage object.'

Reviewer: ```suggestion
Description: 'Removes all keys in a Storage object.'
```

Subjects:
  - 'Data Visualization'
  - 'Web Development'
Tags:
  - 'APIs'
  - 'Browsers'
  - 'Data Types'
  - 'Methods'
  - 'Tuples'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

**.clear()** removes all keys in a Storage object.

## Syntax
```pseudo
Storage.clear()
// return: undefined
```
The method does not accept arguments. It will clear all storage keys and their respective values.

> ## Codebyte Example

Reviewer: ```suggestion
## Example
```

Author: I was under the impression that codebyte examples required the "## Codebyte Example" header and did not require and "output" block? Are codebyte examples only used when a regular Example block is already present?

Reviewer: The problem here is that the code as written doesn't work with the engine running the codebytes.

Author: Is there a way to check what the md file looks like once it goes online? I don't know how to display markdown files on my local machine without seeing them in github. Are there any programs that support markdown?

Reviewer: > Is there a way to check what the md file looks like once it goes online? I don't know how to display markdown files on my local machine without seeing them in github. Are there any programs that support markdown?

@MB1721, there's a list here: https://becomeawritertoday.com/best-markdown-editor/

> New items are added to `sessionStorage` and `localStorage`. The `.clear()` method then removes all data 

Reviewer: ```suggestion
In this example new items are added to `sessionStorage` and `localStorage`. The `.clear()` method then removes all data 
```

from each Storage object.

> ```codebyte/javascript

Reviewer: ````suggestion
```js
````

// Add items to session storage.
sessionStorage.setItem('tab-item1', 'TI-1')
sessionStorage.setItem('tab-item2', 'TI-2')

// Add items to local storage.
localStorage.setItem('domain-item1', 'DI-1')
localStorage.setItem('domain-item2', 'DI-2')

// Remove all items from both local and session storage.
sessionStorage.clear()
localStorage.clear()

// Output storage.
sessionStorage
localStorage

> 
> # Output:
> Storage {length: 0}
> Storage {length: 0}
> ```

Reviewer: ````suggestion
```

 This example results in the following output:

```shell
Storage {length: 0}
Storage {length: 0}
```
````

---

> Title: '.clear()'

Reviewer 2: A folder named clear should be added inside "terms". The path should be -
content/javascript/concepts/storage/terms/clear/clear.md

Description: 'Removes all keys in a Storage object.'
Subjects:
  - 'Data Visualization'
  - 'Web Development'
Tags:
  - 'APIs'
  - 'Browsers'
  - 'Data Types'
  - 'Methods'
  - 'Tuples'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> **.clear()** removes all keys in a Storage object.

Reviewer 2: ```suggestion
**`.clear()`** removes all keys in a `Storage` object.
```

> ## Syntax

Reviewer 2: add an empty line after the syntax header as well as the example header

```pseudo
Storage.clear()

> // return: undefined

Reviewer 2: ```suggestion
```

> ```

Reviewer 2: add an empty line after the block

The method does not accept arguments. It will clear all storage keys and their respective values.

## Example

> In this example new items are added to `sessionStorage` and `localStorage` from console. The `.clear()` method then removes all data 

Reviewer 2: ```suggestion
In this example new items are added to `sessionStorage` and `localStorage` from console. The `.clear()` method then removes all data from each `Storage` object.
```

> In this example new items are added to `sessionStorage` and `localStorage` from console. The `.clear()` method then removes all data 

Reviewer 3: [Proofreading] Always place a comma after an introductory element. 
```suggestion
In this example, new items are added to `sessionStorage` and `localStorage` from console. The `.clear()` method then removes all data 
```

> from each Storage object.

Reviewer 2: ```suggestion
```

```js

> // Add items to session storage.
> sessionStorage.setItem('tab-item1', 'TI-1')
> sessionStorage.setItem('tab-item2', 'TI-2')
> 
> // Add items to local storage.
> localStorage.setItem('domain-item1', 'DI-1')
> localStorage.setItem('domain-item2', 'DI-2')
> 
> // Remove all items from both local and session storage.
> sessionStorage.clear()
> localStorage.clear()
> 
> // Output storage.
> sessionStorage
> localStorage

Reviewer: ```suggestion
// Add items to session storage.
sessionStorage.setItem('tab-item1', 'TI-1');
sessionStorage.setItem('tab-item2', 'TI-2');

// Add items to local storage.
localStorage.setItem('domain-item1', 'DI-1');
localStorage.setItem('domain-item2', 'DI-2');

// Remove all items from both local and session storage.
sessionStorage.clear();
localStorage.clear();

// Output storage.
console.log(sessionStorage.length);
console.log(localStorage.length);
```

```

This example results in the following output:

```shell

> Storage {length: 0}
> Storage {length: 0}

Reviewer: ```suggestion
0
0
```
